### DISCUSSION TITLE
Leave task containers running for real time results?

### DISCUSSION THREAD

**Discussion by ChrisBarefoot** [flyteorg/flyte#1481] [Ideas]
Hey Flyte community.

We're currently playing with Flyte for a proof of concept.

One thing we noticed is that the overhead for small requests of starting and stopping each task's container is significant. Our use case targets real-time responses, if possible.

Is there a concept in flight of leaving those containers up and running and responding to new requests as they come in?

Thanks so much!

**Comment by kumare3** [comment]
Hi @ChrisBarefoot great question. Firstly welcome to the community and thank you for the questions.

So at the moment Flyte restarts a new pod for every task execution. This is to allow multiple versions to co-exist. And as you experienced this can be slow. This is also because Flyte is optimized for longer running tasks where the overhead can be ammortized.

There are a few things we will soon start working on
- Task coalescing, fuse multiple tasks onto the same container. https://github.com/flyteorg/flyte/issues/649
- KNative integration - support light weight functions with low
Spin up overhead. 

But these are slated to be released closer to the end of the year.

As you can imagine there are a few teams interested in the knative plugin

**Reply by ChrisBarefoot** [reply]
Thanks so much @kumare3 for the quick response. We'll take a look at both of those.

**Comment by georgesnelling** [comment]
Hi Chris,

Just curious, what is your high-level use case?  If that's confidential I
completely understand.

-George

**Reply by ChrisBarefoot** [reply]
Thanks George. At a high level you could imagine a text processing pipeline with a mix of both functional and ML-based tasks. 

Something like 
Tokenize -> Lemmatize -> N-Gram Generation -> Vectorization -> ML Scoring.

The runtime aspect of it comes in use cases like chatbots where we want quick response on small inputs of data.

Thanks!

**Reply by kumare3** [reply]
@ChrisBarefoot I completely understand the usecase and we would love to discuss more if your are open to it and talk about it on the slack channel or over vc?

**Reply by ChrisBarefoot** [reply]
Thanks @kumare3. We're still exploring, but will certainly reach out as we have questions. Appreciate it!

**Reply by kumare3** [reply]
@ChrisBarefoot There is a lot of interest for this in the community - and we are looking into options. 
cc @JakeNeyer, Donald MacDonald (github handle missing), @EngHabu so we would love to collaborate if your team is interested as well

**Reply by AdrianoKF** [reply]
Commenting so I'll get notifications (and as a stand-in for Donald MacDonald, since he's my manager). Yes, we'd be quite interested in this use case and possible solutions as well (the Knative and task coalescing approaches mentioned in the other thread sound promising)

**Reply by georgesnelling** [reply]
+ @hamersaw since he's already hip-deep in propeller optimization.
